### PR TITLE
Cleanup cyclomatic complexity with function JoinCluster

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Clear functions with cyclomatic complexity greater than 15.

JoinCluster() exceeded the limit:

https://github.com/karmada-io/karmada/blob/c37bedc1cfe5a98b47703464fed837380c90902f/pkg/karmadactl/join.go#L179

**Which issue(s) this PR fixes**:
Part of #460

### CODE REVIEW

File: pkg/karmadactl/join.go [hunk 1]
```diff
@@ -277,32 +310,19 @@ func JoinCluster(controlPlaneRestConfig, clusterConfig *rest.Config, clusterName
...
-			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(cluster, clusterResourceKind),
-			},
-		},
-	}
-
-	err = util.PatchSecret(controlPlaneKubeClient, secretNamespace, secretName, types.MergePatchType, patchSecretBody)
-	if err != nil {
-		klog.Errorf("failed to patch secret %s/%s, error: %v", secret.Namespace, secret.Name, err)
-		return err
```
Comment (RainbowMango) [inline pkg/karmadactl/join.go:302]: I can't remember why we need to patch the owner reference. 
Why not add the reference when creating the secrets in #67 ?. @mrlihanbo

**Comment by RainbowMango** [comment]
@XiShanYongYe-Chang 
Which test you've done with this patch?

**Comment by XiShanYongYe-Chang** [comment]
> @XiShanYongYe-Chang
> Which test you've done with this patch?

We have join/unjoin e2e test to cover it.

**Pull request merged**